code/modules/spells/spell_types/curse.dm

```diff
@@ -0,0 +1,29 @@
+/proc/curse_of_madness(mob/user, message)
+	if(user) //in this case either someone holding a spellbook or a badmin
+		to_chat(user, "<span class='warning'>You sent a curse of madness with the message \"[message]\"!</span>")
+		message_admins("[ADMIN_LOOKUPFLW(user)] sent a curse of madness with the message \"[message]\"!")
+		log_game("[key_name(user)] sent a curse of madness with the message \"[message]\"!")
+
+	for(var/mob/living/carbon/human/H in GLOB.player_list)
+		var/turf/T = get_turf(H)
+		if(T && !is_station_level(T.z))
+			continue
+		if(H.stat == DEAD)
+			continue
+		if(H.anti_magic_check(TRUE, FALSE))
+			continue
+		if(istype(H.get_item_by_slot(SLOT_HEAD), /obj/item/clothing/head/foilhat))
```

Comment:
It might be funny to add a notification here that the users tinfoil hat protected them from something.

"And they called you crazy! Well who's laughing now?"

Reply:
good idea, i'll add that

Reply:
No benefit in doing it before a second tinfoil hat equil existed that also isn't a subtype of the above (I personally don't think that'll ever happen) unless there's some slight performance benefit.